dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/ConsentAwareStorage.kt

```diff
@@ -124,19 +125,20 @@ internal class ConsentAwareStorage(
     }
 
     /** @inheritdoc */
-    @WorkerThread
+    @AnyThread
     override fun dropAll() {
-        synchronized(lockedBatches) {
-            lockedBatches.forEach {
-                deleteBatch(it, RemovalReason.Flushed)
+        executorService.submitSafe("ConsentAwareStorage.dropAll", internalLogger) {
```

Comment:
same question as above ? What if we call this from an unit test `tearDown` or `setup` method ?

Reply:
> [!IMPORTANT]
> Honestly, if we need to call this from a test, we can adapt the test to this call being async. We should not make things synchronous at runtime because of testing constraints.